File: templates/javascript/v1/strategy/greeterStrategies.js
export const friendlyStrategy = {
  format(name) {
    return `Hey ${name}, great to meet you.`;
  }
};

export const formalStrategy = {
  format(name) {
    return `Hello ${name}. It is a pleasure to meet you.`;
  }
};

export const uppercaseStrategy = {
  format(name) {
    return `HELLO ${name} FROM STRATEGY`.toUpperCase();
  }
};

export class GreeterContext {
  constructor(strategy) {
    this.strategy = strategy;
  }

  setStrategy(strategy) {
    this.strategy = strategy;
  }

  greet(name) {
    return this.strategy.format(name);
  }
}

File: templates/python/v1/strategy/greeter_strategies.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Protocol


class GreeterStrategy(Protocol):
    def format(self, name: str) -> str:
        ...


@dataclass(slots=True)
class FriendlyStrategy:
    def format(self, name: str) -> str:
        return f"Hey {name}, great to meet you."


@dataclass(slots=True)
class FormalStrategy:
    def format(self, name: str) -> str:
        return f"Hello {name}. It is a pleasure to meet you."


@dataclass(slots=True)
class UppercaseStrategy:
    def format(self, name: str) -> str:
        return f"HELLO {name} FROM STRATEGY".upper()


@dataclass(slots=True)
class GreeterContext:
    strategy: GreeterStrategy

    def set_strategy(self, strategy: GreeterStrategy) -> None:
        self.strategy = strategy

    def greet(self, name: str) -> str:
        return self.strategy.format(name)
